torchci/lib/jobUtils.ts

```diff
@@ -133,6 +133,46 @@ LIMIT
   return results !== undefined && results.length === 1 ? results[0].email : "";
 }
 
+export async function hasS3Log(job: RecentWorkflowsData): Promise<boolean> {
+  // This is to handle the infra flaky issue where the log is not available on
+  // S3 and no failure is found
+  const url = `https://ossci-raw-job-status.s3.amazonaws.com/log/${job.id}`;
+  const res = await fetch(url, { method: "HEAD" });
+  return res.status !== 404;
+}
+
+export async function backfillMissingLog(
+  owner: string,
+  repo: string,
+  job: RecentWorkflowsData
+): Promise<boolean> {
+  // This creates a mock GitHub workflow_job completion event to reupload the log
```

Comment:
Good catch, you're right.  I guess the fix should be to download the json from S3 first instead of mocking it like this.  I cost another S3 query, but that's probably ok

Reply:
I think I will need to fix this upstream in the lambda code.

* Part of the event body can be downloaded from S3, but not the full event JSON which is keyed by a random uuid https://github.com/pytorch/test-infra/blob/main/aws/lambda/github-status-test/lambda_function.py#L85-L89
* I could download the log using GitHub API, but then uploading it to S3 requires credential and complex setup.

So, the easier approach would be having a backfill flag on the lambda.

Reply:
I use a mock special action `backfill` here to tell the lambda that this is a backfill request.  The usual event action from GitHub is `completed` https://github.com/pytorch/test-infra/blob/main/aws/lambda/github-status-test/lambda_function.py#L54

Reply:
Here is the lambda update https://github.com/pytorch/test-infra/pull/4740